mapie/classification.py

```diff
@@ -979,91 +367,44 @@ def _check_fit_parameter(
-            train/test set.
-            By default ``None``.
-
-        size_raps: : Optional[float]
-            Percentage of the data to be used for choosing lambda_star and
-            k_star for the RAPS method.
-
-        Returns
-        -------
-        Tuple[NDArray, NDArray, NDArray, NDArray, Optional[NDArray],
-        Optional[NDArray]]
-            - NDArray of shape (n_samples, n_features)
-            - NDArray of shape (n_samples,)
-            - NDArray of shape (n_samples,)
-            - NDArray of shape (n_samples,)
-            - NDArray of shape (n_samples,)
-            - NDArray of shape (n_samples,)
-        """
-        # Split data for raps method
-        raps_split = StratifiedShuffleSplit(
-            n_splits=1, test_size=size_raps, random_state=self.random_state
+        cs_estimator = check_classification_conformity_score(
+            conformity_score=self.conformity_score,
+            method=self.method,
         )
-        train_raps_index, val_raps_index = next(raps_split.split(X, y_enc))
-        X, self.X_raps, y_enc, self.y_raps = (
-            _safe_indexing(X, train_raps_index),
-            _safe_indexing(X, val_raps_index),
-            _safe_indexing(y_enc, train_raps_index),
-            _safe_indexing(y_enc, val_raps_index),
+        # TODO test size_raps depreciated
+        cs_estimator.set_external_attributes(
+            cv=self.cv,
+            classes=self.classes_,
+            label_encoder=self.label_encoder_,
+            size_raps=size_raps,
+            random_state=self.random_state
         )
 
-        # Decode y_raps for use in the RAPS method
-        self.y_raps_no_enc = self.label_encoder_.inverse_transform(self.y_raps)
-        y = self.label_encoder_.inverse_transform(y_enc)
+        # Cast
+        X, y_enc, y = cast(NDArray, X), cast(NDArray, y_enc), cast(NDArray, y)
+        sample_weight = cast(NDArray, sample_weight)
+        groups = cast(NDArray, groups)
 
-        # Cast to NDArray for type checking
-        y_enc = cast(NDArray, y_enc)
-        n_samples = _num_samples(y_enc)
-        if sample_weight is not None:
-            sample_weight = cast(NDArray, sample_weight)
-            sample_weight = sample_weight[train_raps_index]
-        if groups is not None:
-            groups = cast(NDArray, groups)
-            groups = groups[train_raps_index]
+        X, y, y_enc, sample_weight, groups = \
+            cs_estimator.split_data(X, y, y_enc, sample_weight, groups)
+        self.n_samples_ = cs_estimator.n_samples_
 
-        return X, y_enc, y, n_samples, sample_weight, groups
+        check_target(cs_estimator, y)
+
+        return (
+            estimator, cs_estimator, cv, X, y, y_enc, sample_weight, groups
+        )
 
     def fit(
         self,
         X: ArrayLike,
         y: ArrayLike,
         sample_weight: Optional[ArrayLike] = None,
-        size_raps: Optional[float] = 0.2,
+        size_raps: Optional[float] = None,
```

Comment:
why did you change this value ?

Reply:
To give priority to the RAPS class. If I forgot to put a deprecatedwarning, I have to add it

Reply:
Open question : should this value be handled directly by the conformity score ?

Reply:
We would like MapieClassifier to no longer have a dependency on RAPS. Then yes, this value shoud be handled directly by the conformity score.